python/ray/data/tests/test_numpy_support.py

```diff
@@ -22,8 +22,8 @@ def do_map_batches(data):
 
 
 def assert_structure_equals(a, b):
-    assert type(a) == type(b), (type(a), type(b))
-    assert type(a[0]) == type(b[0]), (type(a[0]), type(b[0]))  # noqa: E721
+    assert type(a) is type(b), (type(a), type(b))
+    assert type(a[0]) is type(b[0]), (type(a[0]), type(b[0]))
```

Comment:
I found that for rule E721, the behavior of old and new version of flake8 are different. So maybe add back the `# noqa: E721` comment, and then add a TODO comment to say that this `# noqa` should be removed after flake8 is upgraded. For example: `TODO: Remove noqa after flake8 is upgraded to 7.1.1`

Failed CI: https://buildkite.com/ray-project/microcheck/builds/6190#0192a009-91f2-402b-b318-d7a5fdc04a24

Old version of flake8:

![image](https://github.com/user-attachments/assets/7cf1c1d9-749b-4540-9bc9-c0a308742a35)

New version (7.1.1) of flake8:

![image](https://github.com/user-attachments/assets/48fd70ae-3fc2-4459-a8ff-5d1d7e9f5908)